src/DataShare/DataShare/Share/GetAzDataShareProviderShareSubscription.cs

```diff
@@ -0,0 +1,159 @@
+    using Microsoft.Azure.Management.DataShare;
+    using Microsoft.Azure.Management.DataShare.Models;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Extensions;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Models;
+    using Microsoft.Rest.Azure;
+
+    /// <summary>
+    /// Defines Get-AzDataShareProviderShareSubscription cmdlet.
+    /// </summary>
+    [Cmdlet(
+         "Get",
+         Commands.ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataShareProviderShareSubscription", DefaultParameterSetName = ParameterSetNames.FieldsParameterSet),
+     OutputType(typeof(PSDataShareProviderShareSubscription))]
+    public class GetAzDataShareProviderShareSubscription : AzureDataShareCmdletBase
+    {
+        /// <summary>
+        /// The resource group name of the account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group of the Azure DataShare Account.",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// The Data Share account name of the share.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Azure DataShare Account name.",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Account, "ResourceGroupName")]
+        public string AccountName { get; set; }
+
+        /// <summary>
+        /// The name of share in account
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "share name.",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Share, "ResourceGroupName", "AccountName")]
+        public string ShareName { get; set; }
+
+        /// <summary>
+        /// The share subscription id of the provider share subscription.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The share subscription id of the provider share subscription",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The share subscription id of the provider share subscription",
+            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
```

Comment:
Why is this in the ResourceId parameter set?  Same question for all of these ShareSubscription cmdlets